staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -206,25 +206,32 @@ func newTimeBucketWatchers(clock clock.Clock, bookmarkFrequency time.Duration) *
 // adds a watcher to the bucket, if the deadline is before the start, it will be
 // added to the first one.
 func (t *watcherBookmarkTimeBuckets) addWatcher(w *cacheWatcher) bool {
-	// note that the returned time can be before t.createTime,
-	// especially in cases when the nextBookmarkTime method
-	// give us the zero value of type Time
-	// so buckedID can hold a negative value
-	nextTime, ok := w.nextBookmarkTime(t.clock.Now(), t.bookmarkFrequency)
-	if !ok {
+	wantsBookmark, bucketID := t.calculateNextBookmarkBucketID(w)
```

Comment:
Also - can you please move the refactoring you did here to a separate PR?

I really prefer having bug fixes sit in their own PRs.